var/spack/repos/builtin/packages/automake/package.py

```diff
@@ -39,10 +39,20 @@ def determine_version(cls, exe):
         return match.group(1) if match else None
 
     def patch(self):
+        # Fix a complaint for an unescaped left brace in a regex by perl@5.26(e.g.RHEL8-based OS):
+        if (
+            self.spec.version < Version("1.15.1")
+            and self.spec["perl"].version < Version("5.28")
+            and self.spec["perl"].version >= Version("5.26")
+        ):
```

Comment:
Possibly more legible and expressive:
```suggestion
        if self.spec.satisfies("@:1.15.0 ^perl@5.26:5.27"):
```